src/python/grpcio/grpc/__init__.py

```diff
@@ -1251,6 +1251,24 @@ def set_details(self, details):
         """
         raise NotImplementedError()
 
+    @abc.abstractmethod
```

Comment:
`@abc.abstractmethod` is not backward compatible. We don't want to break existing custom implementations of ServiceContext by our users.